File: hello-php/src/Client/Core/ChunkManager.php
<?php

namespace AudioStreamClient\Core;

/**
 * Chunk manager for handling file chunking operations.
 * Provides utilities for splitting and reassembling data.
 */
class ChunkManager
{
    private const DEFAULT_CHUNK_SIZE = 8192; // 8KB

    /**
     * Split data into chunks.
     *
     * @param string $data Data to split
     * @param int $chunkSize Size of each chunk
     * @return array Array of chunks
     */
    public static function splitIntoChunks(string $data, int $chunkSize = self::DEFAULT_CHUNK_SIZE): array
    {
        $chunks = [];
        $offset = 0;
        $dataSize = strlen($data);

        while ($offset < $dataSize) {
            $size = min($chunkSize, $dataSize - $offset);
            $chunks[] = substr($data, $offset, $size);
            $offset += $size;
        }

        return $chunks;
    }

    /**
     * Get the number of chunks for a given data size.
     *
     * @param int $dataSize Total data size
     * @param int $chunkSize Size of each chunk
     * @return int Number of chunks
     */
    public static function getChunkCount(int $dataSize, int $chunkSize = self::DEFAULT_CHUNK_SIZE): int
    {
        return (int)ceil($dataSize / $chunkSize);
    }

    /**
     * Get the default chunk size.
     *
     * @return int Default chunk size
     */
    public static function getDefaultChunkSize(): int
    {
        return self::DEFAULT_CHUNK_SIZE;
    }
}

File: hello-dart/lib/client/core/chunk_manager.dart
// Chunk manager for handling file chunking operations.
// Provides utilities for splitting and reassembling file chunks.
// Matches Java ChunkManager functionality.

import 'dart:typed_data';

/// Chunk manager for file chunking operations.
class ChunkManager {
  static const int defaultChunkSize = 8192; // 8KB

  /// Split data into chunks of specified size.
  static List<Uint8List> splitIntoChunks(Uint8List data, int chunkSize) {
    final chunks = <Uint8List>[];
    int offset = 0;

    while (offset < data.length) {
      final end =
          (offset + chunkSize < data.length) ? offset + chunkSize : data.length;
      chunks.add(data.sublist(offset, end));
      offset = end;
    }

    return chunks;
  }

  /// Reassemble chunks into a single data buffer.
  static Uint8List reassembleChunks(List<Uint8List> chunks) {
    final totalSize = chunks.fold<int>(0, (sum, chunk) => sum + chunk.length);
    final result = Uint8List(totalSize);

    int offset = 0;
    for (final chunk in chunks) {
      result.setRange(offset, offset + chunk.length, chunk);
      offset += chunk.length;
    }

    return result;
  }

  /// Calculate the number of chunks needed for a given file size.
  static int calculateChunkCount(int fileSize, int chunkSize) {
    return (fileSize + chunkSize - 1) ~/ chunkSize;
  }
}
